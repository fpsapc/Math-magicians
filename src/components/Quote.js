import React, { useEffect, useState } from 'react';

const Quote = () => {
  const [quote, setQuote] = useState([]);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(null);

  useEffect(() => {
    const fetchQuote = async () => {
      const url = 'https://api.api-ninjas.com/v1/quotes?category=love';
      const headers = { 'X-Api-Key': 'aL4yRrpcK1ykMPWPi+rSeA==TEXP1k8ppWk6Zi9w' };
      try {
        const response = await fetch(url, { headers });
        if (!response.ok) {
          throw new Error('Network response was not ok');
        }
        const result = await response.json();
        setQuote(result[0]);
        setLoading(false);
      } catch (error) {
        setError(error.message);
        setLoading(false);
      }
    };
    fetchQuote();
  }, []);

  if (loading) {
    return <div>Loading...</div>;
  }

  if (error) {
    return (
      <div>
        Error:
        {error}
      </div>
    );
  }
  return (
    <div className="quoteContainer">
      <p className="quote">
        Quote of the day:
        {quote.quote}
      </p>
      <p className="author">
        Author:
        {quote.author}
      </p>
    </div>
  );
};

export default Quote;
