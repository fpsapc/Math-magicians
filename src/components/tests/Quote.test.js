import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from 'react-test-renderer';
import Quote from '../Quote';

it('render correctly and mathes the snapshot', () => {
  const tree = renderer.create(<Quote />).toJSON();
  expect(tree).toMatchSnapshot();
});
