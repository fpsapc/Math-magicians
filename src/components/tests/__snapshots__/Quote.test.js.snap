// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly and mathes the snapshot 1`] = `
<div>
  Loading...
</div>
`;
